lib/tasks_screen/models/task.dart

```diff
@@ -0,0 +1,60 @@
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:lachies_life_planner/tasks_screen/models/invalid_task_exception.dart';
+
+class Task {
+  final String id;
+  final String name;
+  final Timestamp dateCreated;
+  final bool isComplete;
+
+  Task({this.id, this.name, this.dateCreated, this.isComplete});
+
+  @override
+  String toString() {
+    return 'Task(${toJson()}';
+  }
+
+  static bool jsonMatchesTaskStructure(Map<String, dynamic> json) =>
+      json.containsKey('id') &&
+      json.containsKey('name') &&
+      json.containsKey('dateCreated') &&
+      json.containsKey('isComplete') &&
+      json.keys.length == 4;
+
+  static Task fromJson(Map<String, dynamic> json) {
+    if (!jsonMatchesTaskStructure(json)) {
+      throw InvalidTaskException(json);
```

Comment:
Should add 'invalid value type' validation here too.  Then I can drop the try/catch

Reply:
Done